src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -740,24 +812,41 @@ export class ExchangeWrapper extends ContractWrapper {
-                                        makerTokenAmount: BigNumber.BigNumber): Promise<boolean> {
+                                        makerTokenAmount: BigNumber.BigNumber,
+                                        exchangeContractAddress: string): Promise<boolean> {
         await assert.isUserAddressAvailableAsync(this._web3Wrapper);
-        const exchangeInstance = await this._getExchangeContractAsync();
+        const exchangeInstance = await this._getExchangeContractAsync(exchangeContractAddress);
         const isRoundingError = await exchangeInstance.isRoundingError.call(
             takerTokenAmount, fillTakerAmount, makerTokenAmount,
         );
         return isRoundingError;
     }
-    private async _getExchangeContractAsync(): Promise<ExchangeContract> {
-        if (!_.isUndefined(this._exchangeContractIfExists)) {
-            return this._exchangeContractIfExists;
+    private async _getExchangeContractAsync(exchangeContractAddress: string): Promise<ExchangeContract> {
+        if (!_.isUndefined(this._exchangeContractByAddress[exchangeContractAddress])) {
+            return this._exchangeContractByAddress[exchangeContractAddress];
         }
+        const ExchangeArtifacts = this._getExchangeArtifactsByAddressOrThrow(exchangeContractAddress);
         const contractInstance = await this._instantiateContractIfExistsAsync((ExchangeArtifacts as any));
-        this._exchangeContractIfExists = contractInstance as ExchangeContract;
-        return this._exchangeContractIfExists;
+        this._exchangeContractByAddress[exchangeContractAddress] = contractInstance as ExchangeContract;
+        return this._exchangeContractByAddress[exchangeContractAddress];
+    }
+    private _getExchangeArtifactsByAddressOrThrow(exchangeContractAddress: string): ContractArtifact {
+        const exchangeArtifacts = _.values<ContractArtifact>(ExchangeArtifactsByName);
+        for (const exchangeArtifact of exchangeArtifacts) {
+            const networkSpecificExchangeArtifactValues = _.values(exchangeArtifact.networks);
+            const exchangeAddressesInArtifact = _.map(
+                networkSpecificExchangeArtifactValues,
+                networkSpecificExchangeArtifact => networkSpecificExchangeArtifact.address,
+            );
+            if (_.includes(exchangeAddressesInArtifact, exchangeContractAddress)) {
+                return exchangeArtifact;
+            }
+        }
+        throw new Error(ZeroExError.EXCHANGE_CONTRACT_DOES_NOT_EXIST);
     }
-    private async _getZRXTokenAddressAsync(): Promise<string> {
-        const exchangeInstance = await this._getExchangeContractAsync();
-        return exchangeInstance.ZRX.call();
+    private async _getZRXTokenAddressAsync(exchangeContractAddress: string): Promise<string> {
+        const exchangeInstance = await this._getExchangeContractAsync(exchangeContractAddress);
+        const ZRXtokenAddress = exchangeInstance.ZRX.call();
```

Comment:
Aren't call requests also async? Shouldn't this be: `await exchangeInstance.ZRX.call()`?